### PR TITLE
Add `allow_failure` annotation to allow failed runs to be passed downstream

### PR DESCRIPTION
**Pull request by madkinsz**
Closes https://github.com/PrefectHQ/prefect/issues/7098

### Example

```python
@task
def fails():
    raise ValueError("Fail task!")

@task
def bar(y):
    return y

@flow
def test_flow():
    f = fails.submit()

    # Previously, `bar` would fail with a `NotReady` state
    # Now, it will receive an exception
    b = bar.submit(allow_failure(f))
    assert isinstance(b.result(), ValueError)
    
    # Previously, `bar` would fail with a `NotReady` state
    # Now, it will run after the upstream completes
    b = bar.submit(1, wait_for=[allow_failure(f)])
    assert b.result() == 1
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Comment by madkinsz** [comment]
I'd like to add some documentation before merging, but otherwise this is pretty straightforward.

**Review by EmilRex — APPROVED** [review]
🚀

**Comment by madkinsz** [comment]
@tpdorsey do you know where documentation would best fit for this?

File: src/prefect/utilities/annotations.py [hunk 1]
```diff
@@ -4,14 +4,39 @@
...
 
 class unmapped:
-    """A container that acts as an infinite array where all items are the same
-    value. Used for Task.map where there is a need to map over a single
-    value"""
+    """
+    Wrapper for iterables.
+
+    Indicates that this input should be sent as-is to all runs created during a mapping
+    operation instead of being split.
```
Comment (bunchesofdonald) [inline src/prefect/utilities/annotations.py:11]: Thank you for updating this.

**Review by bunchesofdonald — APPROVED** [review]

**Comment by madkinsz** [comment]
There's no `wait_for` documentation right now. I can write it but I'd like to get this merged and do it as a follow-up.

**Pull request merged**